weblate/checks/source.py

```diff
@@ -91,14 +91,12 @@ def check_component(self, component: Component) -> Iterable[Unit]:
             unit.pk: unit
             for unit in Unit.objects.filter(translation__component=component)
         }
-        checks = self.get_related_checks(pk_to_unit.keys()).select_related(
-            "unit__source_unit"
-        )
+        checks = self.get_related_checks(pk_to_unit.keys())
 
         source_unit_to_translations = defaultdict(set)
         for check in checks:
```

Comment:
Only two values are needed, so there is no need to fetch more:

```suggestion
        for source_unit_id, translation_id in checks.values_list("unit__source_unit_id", "unit__translation_id"):
```